src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/RecordsFetcherFactory.java

```diff
@@ -24,18 +24,48 @@
     /**
      * Returns a records fetcher processor to be used for processing data records for a (assigned) shard.
      *
-     * @return Returns a record fetcher object
+     * @param getRecordsRetrievalStrategy GetRecordsRetrievalStrategy to be used with the GetRecordsCache
+     * @param shardId ShardId of the shard for which the GetRecordsCache is to be returned
+     *                
+     * @return Returns a GetRecordsCache object
      */
-    GetRecordsCache createRecordsFetcher(GetRecordsRetrievalStrategy getRecordsRetrievalStrategy);
+    GetRecordsCache createRecordsFetcher(GetRecordsRetrievalStrategy getRecordsRetrievalStrategy, String shardId);
 
+    /**
+     * This method sets the maximum number of ProcessRecordsInput objects the GetRecordsCache can hold at any give time.
```

Comment:
You don't need to include the set.  For many of the methods I would use something like
> Sets the maximum number of ProcessRecordsInput that the GetRecordsCache can hold, before further requests are blocked.